### DISCUSSION TITLE
Error when exporting ava as custom package and import into another package.

### DISCUSSION THREAD

**Discussion by johnsonsamuel** [avajs/ava#3084] [Q&A]
So I have a repo(package) where I have exported `ava`  so that I can import this package to other repos for test.

When I tried to import the package and execute the test, I get the below error. Any thoughts?

```
Uncaught exception in index.spec.js

  AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

    assert(refs.runnerChain)
```

#### package name - `ava-typescript-basic`

File content to expose for other repos,

```js
import test from 'ava';

export default test;

```

#### File content of repo where I am importing the package where I expose ava 

```js
import test from 'ava-typescript-basic';

test('hello', t => {
    t.pass();
})
```

but when I run the test, I get the below error ❌

```js

 npm run test

> test
> ava


  Uncaught exception in index.spec.js

  AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

    assert(refs.runnerChain)


  ✖ index.spec.js exited with a non-zero exit code: 1
  ─

  1 uncaught exception
```

Any thoughts?

**Comment by novemberborn** [comment]
My guess is that the AVA version in your package is different from the one in your repository. Try with `npm ls ava` to confirm. When you launch AVA from the CLI, it provides state in the test workers that need to be accessed by the exported functions. If you have different AVA versions or installations it won't work.

**Comment by stefiix92** [comment]
I've run into similar issues. I'm pretty sure both projects reference the `4.3.3` ava version and still causing troubles. I have been using the `3.15.0` version for a while, now I want to bump up to the latest.

**Comment by SimTheFool** [comment]
Hey ! Have you found a solution ? I'm running into a similar issue @stefiix92 @johnsonsamuel

**Reply by stefiix92** [reply]
Hi, I exported the shared functions from the common folder this way:

```
export function corsTests(test) {
  test("OPTIONS Cors Is Missing", async (t) => {
    ...
    t.is(res.status, 200);
  });
....
```

and then I reference shared tests by:
```
import {corsTests} from 'common/test/cors.test.js'

test.beforeEach((t) => { ... }
corsTests(test) // pass test as a test function
```

**Comment by rglover** [comment]
May be able to shed some light and help here. I'm writing two separate packages which reference `ava` as a dependency (@joystick.js/cli and @joystick.js/test). In one package, I do a straight `import test from 'ava';` and in another, I call to `/node_modules/.bin/ava` via a `child_process.exec()`.

I was getting the AssertionError described above and the workaround was to make sure that I was always referencing the exact same `ava` install path. So, in the project where I'm installing those packages, make sure both packages reference _that project's_ `node_modules/ava` (versus referencing the `node_modules/ava` installed in their own package source).

**Reply by novemberborn** [reply]
Marking it as a peer dependency in both packages can work for that.
